cmd/installer/main.go

```diff
@@ -17,66 +19,71 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-func parseOptions(uninstall *bool, options *install.SafeOptions) *rest.Config {
-	var etcdCA, etcdCert, etcdKey, tlsCA, tlsCert, tlsKey string
-	var logLevel string
-	var pullPolicy string
-	flag.BoolVar(&options.Network.ShouldUseHost, "network-host", false, "Use network host")
-	flag.StringVar(&options.OptionsCommon.Namespace, "namespace", "docker", "Namespace in which to deploy")
-	flag.DurationVar(&options.OptionsCommon.ReconciliationInterval, "reconciliation-interval", constants.DefaultFullSyncInterval, "Interval of reconciliation loop")
-	flag.StringVar(&options.OptionsCommon.Tag, "tag", "latest", "Image tag")
-	flag.StringVar(&pullPolicy, "pull-policy", string(corev1types.PullAlways), fmt.Sprintf("Image pull policy (%q|%q|%q)", corev1types.PullAlways, corev1types.PullNever, corev1types.PullIfNotPresent))
-	flag.StringVar(&options.Etcd.Servers, "etcd-servers", "", "etcd server addresses")
-	flag.BoolVar(&options.SkipLivenessProbes, "skip-liveness-probes", false, "Disable liveness probe on Controller and API server deployments. Use this when HTTPS liveness probe fails.")
+func main() {
+	if err := do(); err != nil {
+		fmt.Fprintf(os.Stderr, "Installation failed: %s", err)
+		os.Exit(1)
+	}
+}
 
-	flag.StringVar(&etcdCA, "etcd-ca-file", "", "CA of etcd TLS certificate")
-	flag.StringVar(&etcdCert, "etcd-cert-file", "", "TLS client certificate for accessing etcd")
-	flag.StringVar(&etcdKey, "etcd-key-file", "", "TLS client private key for accessing etcd")
+func do() error {
```

Comment:
nit: Not a huge fan of `do` for a function name but can't think of anything better given how it's been factored.